util.py

```diff
@@ -0,0 +1,108 @@
+
+import re as regex
+from glob import iglob
+from os.path import join
+from typing import Generator
+
+import discord
+from discord.ext import commands
+
+# ---------------------> Utility functions
+
+# Wraps around commands to split args into flags and params.
+#   - func MUST follow async (self, ctx, flags, params) -> Any
+#   - decorator should be placed below @bot.command() decorator
+
+def extract_flags(blacklist: list[str] | None = None, whitelist: list[str] | None = None, **thesaurus: dict[str, list[str]]):
+    def wrapper(func):
+        async def wrapped(self, ctx, *args, **kwargs):
+            flags  = []
+            params = []
+
+            for arg in list(args):
+
+                # Parse flag
+                if arg.startswith('-'):
+                    flag = arg[1:]
+
+                    # Translate synonyms into default
+                    if flag not in thesaurus.keys():
+                        for default, synonyms in thesaurus.items():
+                            if flag in synonyms:
+                                flag = default
+                                break
+                    
+                    # Filter blacklisted flags
+                    if blacklist != None and flag in blacklist:
+                        continue
+
+                    # Filter non-whitelisted flags
+                    if whitelist != None and flag not in whitelist:
+                        continue
+
+                    flags.append(flag)
+                
+                # Parse parameter
+                else:
+                    params.append(arg)
+
+            return await func(self, ctx, flags, params)
+        return wrapped
+    return wrapper
+
+# Yields all extension files in path.
+#   - path contains path to extensions                      default is 'extensions'
+#   - prefix_path toggles prefixing with extension path     default is False
+#   - recursive toggles recursive search                    default is True
```

Comment:
Eh it wasnt my intention, I just thought for global util functions it might be good to resolve ambiguity. But I mean I'm willing to do it for everything, although in that case we probably should use proper docstrings that editors actually understand. Is there a standard that both vs and pycharm understand?